File: 2024/Feb 2024/2-14-24/Parameter Array Order/RearrangeArrayElementsBySign_2149.py
from typing import List

class Solution:
   def rearrangeArray(self, nums: List[int]) -> List[int]:
      posIndex = 0
      negIndex = 1
      solution = [0] * len(nums)

      for curNum in nums:
         if curNum > 0:
            solution[posIndex] = curNum
            posIndex += 2
         else:
            solution[negIndex] = curNum
            negIndex += 2

      return solution

File: 2024/February/2-14 2149 Rearrange Array Elements By Sign/Parameter Array Order/RearrangeArrayElementsBySign_2149.java
public class RearrangeArrayElementsBySign_2149 {
   public static void main(String args[]) {
      int[] nums = {3, 1, -2, -5, 2, -4};
      for(int i : new Solution().rearrangeArray(nums)) {
         System.out.print(i + " ");
      }  
   }
}

class Solution {
   public int[] rearrangeArray(int[] nums) {
      /* Initialize variables */
      int[] solution = new int[nums.length];
      int posIndex = 0;
      int negIndex = 1;
      
      /* Iterate through nums */
      for(int i : nums) {
         if(i > 0) {   // If current num is positive, place into next pos index
            solution[posIndex] = i;
            posIndex += 2;
         }
         else {   // If current num is negative, place into next neg index
            solution[negIndex] = i;
            negIndex += 2;
         }
      }  
   

      return solution;
   }
}